src/App.js

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react';
+import './App.css';
+import Grid from './Grid';
+import getNextPosition from './Path';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      snake: [40],
+      food: 13,
+      currentDirection: 'right',
+      newDirection: 'right',
+      score: 0,
+      timer: 0,
+      message:''
+    };
+    this.moveSnake = this.moveSnake.bind(this);
+    this.startGame = this.startGame.bind(this);
+    this.getKeyDirection = this.getKeyDirection.bind(this);
+  }
+  getKeyDirection(keycode) {
+    this.setState({message : ''});
+    let direction;
+    let old = this.state.newDirection;
+    switch (event.keyCode) {
+      case 37:
+        direction = 'left';
+        break;
+      case 38:
+        direction = 'up';
+        break;
+      case 39:
+        direction = 'right';
+        break;
+      case 40:
+        direction = 'down';
+        break;
+      default:
+        direction = this.state.newDirection;
+        break;
+    };
+    this.setState({ currentDirection: old, newDirection: direction});
+  }
+
+  moveSnake() {
+    const currentDirection = this.state.currentDirection;
+    const newDirection = this.state.newDirection;
+    const snake = this.state.snake;
+    // Gets the last element in the array
+    const lastPostion = snake[snake.length - 1];
+    const nextPosition = getNextPosition(currentDirection, newDirection, lastPostion);
+    this.updateSnake(nextPosition);
+    this.setState({timer : this.state.timer + 1});
+    // Checks if snake hits itself
+    if ((new Set(snake)).size !== snake.length) {
```

Comment:
This might lead to a very long game as the snake would be longer than itself and still not game over for the player